core/modules/map.rs

```diff
@@ -146,6 +146,14 @@ pub(crate) struct ModuleMap {
 }
 
 impl ModuleMap {
+  /// There is a circular Rc reference between the module map and the futures,
+  /// so when destroying the module map we need to clear the pending futures.
+  pub fn clear_pending_futures(&self) {
+    self.preparing_dynamic_imports.borrow_mut().clear();
+    self.pending_dynamic_imports.borrow_mut().clear();
+    self.code_cache_ready_futs.borrow_mut().clear();
```

Comment:
We should clear all the data before returning this. Just to make sure nothing else holds an `Rc<>` to module map.